app/models/user.rb

```diff
@@ -5,32 +5,56 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
+
   def has_membership_application?
     membership_applications.size > 0
   end
 
+
   def has_company?
     has_membership_application? && (membership_applications.select { |app| app.company }).count > 0
   end
 
+
   def membership_application
     has_membership_application? ? membership_applications.last : nil
   end
 
+
   def company
     has_company? ? membership_application.company : nil
   end
 
+
   def admin?
     admin
   end
 
+
   def is_member?
     is_member
   end
 
+
   def is_member_or_admin?
     admin? || is_member?
   end
 
+
+  def is_in_company_numbered?(company_num)
+    is_member? && !(companies.detect { |c| c.company_number == company_num }).nil?
```

Comment:
Yes, you're right that `companies` is an array - except when the first condition is met (`if admin?`), in which case an `ActiveRecord_Relation` is returned.  So, how about this:

1) In `companies` method, return `Company.all.to_a` if `admin?` is true.  So, now we're assured that we get the same object type (array) from that method.

2) Line 45 in this method could then be:
```
is_member? && companies.map(&:company_number).include? company_num
```